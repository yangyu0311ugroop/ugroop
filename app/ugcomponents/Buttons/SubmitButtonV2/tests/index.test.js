import { DO_NOTHING } from 'appConstants';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { SubmitButton } from '../index';

describe('<Button />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
  };

  beforeEach(() => {
    rendered = shallow(<SubmitButton {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(SubmitButton).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('handleClick()', () => {
    it('should call onClick', () => {
      const onClick = jest.fn();
      rendered.setProps({ onClick });

      expect(instance.handleClick('some', 'props')).not.toBe(DO_NOTHING);

      expect(onClick).toBeCalledWith('some', 'props');
    });

    it('should not do anything', () => {
      rendered.setProps({ onClick: undefined });

      expect(instance.handleClick('some', 'props')).toBe(DO_NOTHING);
    });
  });

  describe('renderLoading()', () => {
    it('should renderLoading correctly', () => {
      const snapshot = shallow(<div>{instance.renderLoading()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });

  describe('renderValid()', () => {
    it('should renderValid correctly', () => {
      const snapshot = shallow(<div>{instance.renderValid()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });

  describe('render()', () => {
    it('should render valid', () => {
      instance.renderValid = () => 'renderValid';

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });

    it('should render loading', () => {
      instance.renderLoading = () => 'renderLoading';
      rendered.setProps({ loading: true });

      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
