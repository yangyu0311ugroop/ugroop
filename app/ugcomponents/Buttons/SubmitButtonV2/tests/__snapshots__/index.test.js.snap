// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render() should render loading 1`] = `
<div>
  renderLoading
</div>
`;

exports[`<Button /> render() should render valid 1`] = `
<div>
  renderValid
</div>
`;

exports[`<Button /> renderLoading() should renderLoading correctly 1`] = `
<div>
  <WithStyles(Button)
    color="blue"
    dense={true}
    size="small"
  >
    <WithStyles(Icon)
      className=""
      icon="lnr-refresh2"
    />
    <span>
      <FormattedMessage
        defaultMessage="Loading.."
        id="ShareList.components.InviteByEmail.loading"
      />
    </span>
  </WithStyles(Button)>
</div>
`;

exports[`<Button /> renderValid() should renderValid correctly 1`] = `
<div>
  <WithStyles(Button)
    color="green"
    dense={true}
    disabled={false}
    onClick={[Function]}
    size="small"
    type="submit"
  >
    <WithStyles(Icon)
      icon="lnr-magnifier"
    />
    <span>
      Submit
    </span>
  </WithStyles(Button)>
</div>
`;
